Controller/FolderController.php

```diff
@@ -26,22 +25,10 @@ class FolderController extends Controller
      * Displays the sub folder if it exists
      *
      * @param \eZ\Publish\API\Repository\Values\Content\Location $location of a folder
-     * @throws NotFoundHttpException $location is flagged as invisible
      * @return \Symfony\Component\HttpFoundation\Response
      */
-    public function showFolderListAsideViewAction( Location $location )
+    public function showFolderListAsideViewAction( Location $location, $viewType, $layout = false, array $params = array() )
```

Comment:
`showFolderListAsideViewAction` is used to display second level items in folders tree. We don't want to have recursive actions, all we need is to display only first level subfolders.

You may combine those two actions but be aware to assign some variable that will inform second level request to exit instead of making more (and more...) recursive requests...